src/main/java/org/vanilladb/core/storage/buffer/BufferPoolMgr.java

```diff
@@ -33,7 +34,9 @@
 	private AtomicInteger numAvailable;
 
 	// Optimization: Lock striping
-	private Object[] anchors = new Object[1009];
+	private int stripSize = 1009;
+	private Object[] anchors = new Object[stripSize];
+	private ReentrantLock[] xSwapLocks = new ReentrantLock[stripSize];

```

Comment:
Maybe we can call `anchors` as `fileLocks` and `xSwapLocks` as `blockLocks` because they create critical sections for files and blocks.

Reply:
Great suggestion.
I was stuck at naming for a while